composeml/label_times/object.py

```diff
@@ -254,26 +273,33 @@ def bin(self, bins, quantiles=False, labels=None, right=True):
             2   high
             3    low
         """  # noqa
-        label_times = self.copy()
-        values = label_times[self.label_name].values
+        lt = self.copy()
+        values = lt[self.label_name].values
 
         if quantiles:
-            label_times[self.label_name] = pd.qcut(values, q=bins, labels=labels)
+            values = pd.qcut(values, q=bins, labels=labels, precision=precision)
 
         else:
-            label_times[self.label_name] = pd.cut(values, bins=bins, labels=labels, right=right)
+            if isinstance(bins, list):
+                for i, edge in enumerate(bins):
+                    if edge in ['-inf', 'inf']:
+                        bins[i] = float(edge)
```

Comment:
If you wanted to you can replace these three lines with a list comprehension, although it might not be as readable. Not sure which would have better performance.
```
bins = [float(edge) if edge in ['-inf', 'inf'] else edge for edge in bins]
```

Reply:
I agree. I think it'd make sense to apply the change if there is a non-trivial improvement on performance. Otherwise, I think it'd make sense to use whichever is more readable.

Reply:
I'm guessing we don't expect the `bins` list to ever be very long, so performance probably isn't much of an issue here. The original code is probably a bit more readable in this case.